fdbrpc/include/fdbrpc/LoadBalance.actor.h

```diff
@@ -311,32 +311,48 @@ Future<Void> replicaComparison(Req req,
 			}
 		} else if (!srcLB.present() || !srcLB.get().error.present()) {
 			// Verify that the other SS servers in the team have the same data.
-			state std::vector<Future<Optional<ErrorOr<Resp>>>> restOfTeamFutures;
-			restOfTeamFutures.reserve(ssTeam->size() - 1);
+			std::vector<uint64_t> candidates;
+			// candidates includes all healthy SS endpoints in the team except the one we already
+			// have a response from
 			for (int i = 0; i < ssTeam->size(); i++) {
 				RequestStream<Req, P> const* si = &ssTeam->get(i, channel);
-				if (si->getEndpoint().token.first() !=
-				    srcEndpointId) { // don't re-request to SS we already have a response from
-					if (!IFailureMonitor::failureMonitor().getState(si->getEndpoint()).failed) {
-						resetReply(req);
-						restOfTeamFutures.push_back((
-						    requiredReplicas == BEST_EFFORT
-						        ? timeout(si->tryGetReply(req), FLOW_KNOBS->LOAD_BALANCE_FETCH_REPLICA_TIMEOUT)
-						        : timeout(errorOr(si->getReply(req)), FLOW_KNOBS->LOAD_BALANCE_FETCH_REPLICA_TIMEOUT)));
-					} else if (requiredReplicas == ALL_REPLICAS) {
-						TraceEvent(SevWarnAlways, "UnreachableStorageServer")
-						    .detail("SSID", ssTeam->getInterface(i).id());
-						throw unreachable_storage_replica();
-					}
+				if (si->getEndpoint().token.first() == srcEndpointId) {
+					// Don't re-request to SS we already have a response from
+					continue;
+				}
+				if (!IFailureMonitor::failureMonitor().getState(si->getEndpoint()).failed) {
+					candidates.push_back(si->getEndpoint().token.first());
+				} else if (requiredReplicas == ALL_REPLICAS) {
+					TraceEvent(SevWarnAlways, "UnreachableStorageServer").detail("SSID", ssTeam->getInterface(i).id());
+					throw unreachable_storage_replica();
 				}
 			}
+			int numReplicaToRead = candidates.size();
+			if (requiredReplicas != BEST_EFFORT && requiredReplicas != ALL_REPLICAS) {
+				ASSERT(requiredReplicas > 0);
+				numReplicaToRead = std::min((int)candidates.size(), requiredReplicas);
+			}
+			state std::vector<Future<Optional<ErrorOr<Resp>>>> restOfTeamFutures;
+			restOfTeamFutures.reserve(numReplicaToRead);
+			// Randomly select numReplicaToRead SSes to read from
+			deterministicRandom()->randomShuffle(candidates);
+			candidates.erase(candidates.begin() + numReplicaToRead, candidates.end());
+			std::unordered_set<uint64_t> ssToRead(candidates.begin(), candidates.end());
 
-			if (requiredReplicas == BEST_EFFORT || requiredReplicas == ALL_REPLICAS) {
-				wait(waitForAllReady(restOfTeamFutures));
-			} else {
-				wait(waitForQuorumReplies(&restOfTeamFutures, requiredReplicas));
+			for (int i = 0; i < ssTeam->size(); i++) {
+				RequestStream<Req, P> const* si = &ssTeam->get(i, channel);
+				if (ssToRead.find(si->getEndpoint().token.first()) == ssToRead.end()) {
```

Comment:
[contains](https://en.cppreference.com/w/cpp/container/unordered_set/contains.html) is more readable: 

```suggestion
				if (!ssToRead.contains(si->getEndpoint().token.first())) {
```